src/editor.js

```diff
@@ -722,98 +724,56 @@ export class OdooEditor extends EventTarget {
-        // this hack: we put a fake element acting as a one-space-to-remove
-        // element (like an image) at the selection start position and we
-        // hit backspace until that element is removed.
-        if (pos1[0].nodeType === Node.TEXT_NODE) {
-            // First, if the selection start is in a text node, we have to
-            // split that text node to be able to put the fake element
-            // in-between.
-            const splitNode = pos1[0];
-            const splitOffset = pos1[1];
-            const willActuallySplitPos2 =
-                pos2[0] === splitNode && splitOffset > 0 && splitOffset < splitNode.length;
-            pos1 = [splitNode.parentNode, splitTextNode(splitNode, splitOffset)];
-            if (willActuallySplitPos2) {
-                if (pos2[1] < splitOffset) {
-                    pos2[0] = splitNode.previousSibling;
+        let range = getDeepRange(this.document, { sel, splitText: true, select: true });
+        if (!range) return;
+        // "Correct" triple click.
+        if (!range.endOffset && !range.endContainer.previousSibling) {
+            const previous = previousLeaf(range.endContainer, this.dom);
+            previous && range.setEndAfter(previous);
+        }
+        const start = range.startContainer;
+        let end = range.endContainer;
+        // Let the DOM split and delete the range.
+        const doJoin = closestBlock(start) !== closestBlock(range.commonAncestorContainer);
+        let next = nextLeaf(end, this.dom);
+        range.deleteContents();
+        setCursor(start, nodeSize(start));
+        range = getDeepRange(this.document, { sel });
+        // If the end container was fully selected, extractContents may have
+        // emptied it without removing it. Ensure it's gone.
+        while (end && end !== this.dom && !end.contains(range.endContainer) && !isVisible(end, 1)) {
+            const parent = end.parentNode;
+            end.remove();
+            end = parent;
+        }
+        // Ensure empty blocks be given a <br> child.
+        const block = closestBlock(range.endContainer);
+        if (isBlock(block) && !isVisible(block, true)) {
+            block.appendChild(document.createElement('br'));
+        }
+        // Ensure we trailing space remains visible.
+        const joinWith = range.endContainer;
+        let oldText = joinWith.textContent;
+        doJoin && oldText.endsWith(' ') && (joinWith.textContent = oldText.replace(/ $/, '\u00A0'));
+        // Rejoin blocks that deleteContents may have split in two.
+        while (doJoin && next && !(next.previousSibling && next.previousSibling === joinWith)) {
+            const restore = preserveCursor(this.document);
+            const res = this._protect(() => {
+                next.oDeleteBackward();
+                if (!this.dom.contains(joinWith)) {
+                    this.torollback = 9; // tried to delete too far -> roll it back.
```

Comment:
We are professionnals, please create a new constant for this rollback error type.
Besides, roll it ! 🎲

Reply:
I didn't because I'm using this super locally, in this function exclusively, and didn't want to make it seem like it's a thing and encourage its use elsewhere ^^'

Reply:
Two solutions here IMO. Either we canonize this error in a `DELETED_TOO_FAR_ROLLBACK_CODE` or something like that, which is my first proposition, or we do not use the `torollback` mechanism at all and replace it with a local boolean condition.

Example:
```js
let woopsTooFar = false;
// ... code
if (woopsTooFar || [UNBREAKABLE_ROLLBACK_CODE, UNREMOVABLE_ROLLBACK_CODE].includes(res)) {
...
```

What do you think ?